hypothesis-python/src/hypothesis/internal/conjecture/shrinking/ordering.py

```diff
@@ -79,7 +79,9 @@ def sort_regions_with_gaps(self):
         """
         for i in range(1, len(self.current) - 1):
             if self.current[i - 1] <= self.current[i] <= self.current[i + 1]:
-                continue
+                # This branch is never taken on versions of Python where dicts
+                # preserve their insertion order (pypy or cpython >= 3.7)
```

Comment:
This comment seems unlikely. It's much more plausible we're hitting https://bugs.python.org/issue2506

Reply:
Huh, yeah, and that [changed in CPython 3.7 as well](https://twitter.com/raymondh/status/961309702515605504), which explains why it was covered before.  And it looks like PyPy does the same optimisation, so I'll update the comment.